indico/web/client/js/jquery/widgets/jinja/ckeditor_widget.js

```diff
@@ -5,37 +5,38 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
-/* global RichTextEditor:false, $E:false */
+import ClassicEditor from 'ckeditor';
 
-(function(global) {
-  'use strict';
-
-  global.setupCKEditorWidget = function setupCKEditorWidget(options) {
-    options = $.extend(
-      true,
-      {
-        fieldId: null,
-        simple: false,
-        images: false,
-        height: 475,
-      },
-      options
-    );
+import {getConfig} from 'indico/ckeditor';
 
-    var field = $('#' + options.fieldId);
-    var editor = new RichTextEditor(600, options.height, options.simple, options.images);
-    editor.set(field.val());
-    editor.onLoad(function() {
-      editor.onChange(function() {
-        field.val(editor.get()).trigger('change');
-      });
+(function(global) {
+  global.setupCKEditorWidget = async function setupCKEditorWidget(options) {
+    const {fieldId, images = false, width, height = 475, ...rest} = options;
+    const field = document.getElementById(fieldId);
+    const editor = await ClassicEditor.create(field, {...getConfig({images}), ...rest});
+    editor.setData(field.value);
+    editor.model.document.on('change:data', () => {
+      field.value = editor.getData();
+      field.dispatchEvent(new Event('change', {bubbles: true}));
+    });
+    editor.editing.view.change(writer => {
+      writer.setStyle('width', `${width}px` || 'auto', editor.editing.view.document.getRoot());
+      writer.setStyle('height', `${height}px`, editor.editing.view.document.getRoot());
     });
-    $E(options.fieldId + '-editor').set(editor.draw());
     // Re-position the dialog if we have one since the initial position is
     // wrong due to the editor being loaded after the dialog has been opened.
-    var dialog = field.closest('.ui-dialog-content');
+    const dialog = $(field.closest('.ui-dialog-content'));
```

Comment:
`,closest()` already returns a jquery object, no need for this `$()` around it

Reply:
I'm no longer using jquery here. `document.querySelector().closest` seems to be supported in most modern browsers.

https://caniuse.com/#feat=element-closest

The reason why I'm wrapping `$()` is because we intend to call a jquery dialog afterwards, `$.dialog`. I don't think there's another way to call it.

Reply:
oh, i wasn't aware `field` isn't a jquery object.. nevermind then, also on my other comments i guess